### PR TITLE
Script that turns any YT Link into embed format

### PR DESCRIPTION
**Pull request by yaswant2403**
Wrote a JS to modify given YT link. However, I'm unable to implement the file into the component HTML. I declared a variable and call the function as such
```
// in dashboard.components.ts

declare var getVideoID: any;
// in setLecture()
link = new getVideoID(link)
```
However, this just doesn't generate a preview. Because it doesn't create a preview, lectureData becomes NULL and I'm unable to see what the JSON looks like.

### CODE REVIEW

**Comment by yaswant2403** [comment]
All Resources used:
https://github.com/brandly/angular-youtube-embed
https://jsfiddle.net/ybwtkLf3/105/ (To test out my JS)
https://stackoverflow.com/questions/37081943/angular-2-import-external-js-file-into-component

**Comment by yaswant2403** [comment]
Finished Converting/Rendering different types of Non-Embed Links

File: frontend/src/app/pages/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -24,7 +24,24 @@ export class DashboardComponent implements OnInit {
     }
     this.isSubmit = lectureData
   }
+  getVideoID(text) {
```
Comment (jonahtjandra) [inline frontend/src/app/pages/dashboard/dashboard.component.ts]: Function name should accurately describe what it's trying to do. So, I believe getVideoLink or something like that would be more appropriate.

File: frontend/src/app/pages/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -24,7 +24,24 @@ export class DashboardComponent implements OnInit {
...
+        text = intro.concat(text);
+    }
+    // regex from http://stackoverflow.com/a/5831191/1614967
+    var re = /https?:\/\/(?:[0-9A-Z-]+\.)?(?:youtu\.be\/|youtube?\.com\S*[^\w\s-])([\w-]{11})(?=[^\w-]|$)[?=&+%\w.-]*/ig;
+    var id = text.replace(re,'$1');
+    if (id.includes('#')) {
+        id = id.split('#')[0];
+    }
+    var embed = "https://youtube.com/embed/"
+    id = embed.concat(id)
```
Comment (jonahtjandra) [inline frontend/src/app/pages/dashboard/dashboard.component.ts]: Again, variable name should be more accurate and descriptive.

**Comment by jonahtjandra** [comment]
Good work!

**Pull request merged**